distribution/lib/Standard/Table/0.0.0-dev/src/Excel/Excel_Workbook.enso

```diff
@@ -289,12 +297,50 @@ type Excel_Workbook
 
     ## PRIVATE
     with_java_workbook : (Workbook -> Any) -> Any ! Illegal_State
-    with_java_workbook self f = Illegal_State.handle_java_exception <|
+    with_java_workbook self f = Illegal_State.handle_java_exception <| handle_invalid_location <| Empty_Sheet.handle_java_exception <|
         case self.excel_connection_resource_ref.get of
             Nothing ->
                 Error.throw (Illegal_State.Error "The workbook is being used after it was closed.")
             resource -> resource.with connection->
                 connection.withWorkbook f
 
+    ## GROUP Standard.Base.Input
+       ICON data_input
+       Reads one or more sheets into a Table.
+
+       Arguments:
+       - sheet_names: the names of the sheets to read. Defaults to all sheets.
+         An invalid `sheet_name` will be reported as a problem.
+       - headers: whether to use the first row as headers (default is
+         `Detect_Headers`) for each sheet.
+       - return: how to return the tables. Defaults to `Merged_Table` meaning
+         the tables are merged into a single table.
+       - on_problems: how to handle problems during reading. Defaults to
+         `Report_Warning`.
+    @sheet_names (s-> build_sheet_selector s)
+    read_many : Vector Text -> Headers -> Return_As -> Problem_Behavior -> Table
+    read_many self sheet_names:Vector=self.sheet_names (headers:Headers=..Detect_Headers) (return:Return_As=..Merged_Table) (on_problems:Problem_Behavior=..Report_Warning) =
+        if sheet_names.is_empty then Error.throw (Illegal_Argument.Error "No sheets selected.") else
+            tables = sheet_names.map on_problems=on_problems t-> self.read t headers
+            case return of
+                Return_As.Table_Of_Tables -> Table.new [["SheetName", sheet_names], ["Table", tables]]
+                Return_As.Merged_Table match ->
+                    with_names = tables.zip sheet_names tbl->name-> if tbl == Nothing then Nothing else tbl.set name "SheetName" . reorder_columns ["SheetName"]
+                    Table.from_union (with_names.filter Filter_Condition.Not_Nothing) match keep_unmatched_columns=True
+
 ## PRIVATE
 close_connection c = c.close
+
+## PRIVATE
+build_sheet_selector : Excel_Workbook -> Display -> Widget
+build_sheet_selector workbook:Excel_Workbook display:Display=Display.Always =
```

Comment:
```suggestion
build_sheet_selector workbook:Excel_Workbook display:Display=Display.Always -> Widget =
```

Especially for PRIVATE methods, I feel like there's no reason in having the duplicated signature - I think we should try to move towards preferring the in-line signatures in most cases. 

The only case that comes to my mind that should have the one above is when we want to opt-out from type checks, e.g. because the accepted types are actually broader because we treat the type annotation just like an 'interface hint'.